local-cluster/tests/local_cluster_slow_1.rs

```diff
@@ -386,6 +391,191 @@ fn test_kill_partition_switch_threshold_progress() {
+    let vote_keypair = our_info.info.voting_keypair;
+    let bad_leader_id = cluster.entry_point_info.id;
+    let bad_leader_ledger_path = cluster.validators[&bad_leader_id].info.ledger_path.clone();
+    info!(
+        "our node id: {}, vote id: {}",
+        node_keypair.pubkey(),
+        vote_keypair.pubkey()
+    );
+
+    // 3) Start up a spy to listen for votes
+    let exit = Arc::new(AtomicBool::new(false));
+    let (gossip_service, _tcp_listener, cluster_info) = gossip_service::make_gossip_node(
+        // Need to use our validator's keypair to gossip EpochSlots and votes for our
+        // node later.
+        Keypair::from_bytes(&node_keypair.to_bytes()).unwrap(),
+        Some(&cluster.entry_point_info.gossip),
+        &exit,
+        None,
+        0,
+        false,
+        SocketAddrSpace::Unspecified,
+    );
+
+    let num_votes_simulated = Arc::new(AtomicUsize::new(0));
+    let voter_thread_sleep_ms: usize = 100;
+    let t_voter = {
+        let exit = exit.clone();
+        let num_votes_simulated = num_votes_simulated.clone();
+        // push to tpu
+        let (rpc, tpu) = cluster_tests::get_client_facing_addr(&cluster.entry_point_info);
+        let client = ThinClient::new(rpc, tpu, cluster.connection_cache.clone());
+        let cluster_funding_keypair = cluster.funding_keypair.insecure_clone();
+        std::thread::spawn(move || {
+            let mut cursor = Cursor::default();
+            loop {
+                if exit.load(Ordering::Relaxed) {
+                    return;
+                }
+
+                let (labels, votes) = cluster_info.get_votes_with_labels(&mut cursor);
+                let mut parsed_vote_iter: Vec<_> = labels
+                    .into_iter()
+                    .zip(votes.into_iter())
+                    .filter_map(|(_label, leader_vote_tx)| {
+                        let vote = vote_parser::parse_vote_transaction(&leader_vote_tx)
+                            .map(|(_, vote, ..)| vote)
+                            .unwrap();
+                        // Filter out empty votes
+                        if !vote.is_empty() {
+                            Some((vote, leader_vote_tx))
+                        } else {
+                            None
+                        }
+                    })
+                    .collect();
+
+                parsed_vote_iter.sort_by(|(vote, _), (vote2, _)| {
+                    vote.last_voted_slot()
+                        .unwrap()
+                        .cmp(&vote2.last_voted_slot().unwrap())
+                });
+
+                for (parsed_vote, leader_vote_tx) in &parsed_vote_iter {
+                    if let Some(latest_vote_slot) = parsed_vote.last_voted_slot() {
+                        info!("received vote for {}", latest_vote_slot);
+                        let vote_hash = parsed_vote.hash();
+                        info!(
+                            "Simulating vote from our node on slot {}, hash {}",
```

Comment:
nit: since we do this in a handful of tests now might be worth creating some reusable shim that spies gossip and votes w/ some modification